api/src/main/java/com/exelaration/abstractmemery/services/implementations/MemeServiceImpl.java

```diff
@@ -26,4 +29,20 @@ public Meme save(Meme meme) {
       return null;
     }
   }
+
+  public String getMeme(int id) {
+    Optional<Meme> meme = memeMetadataService.findById(id);
```

Comment:
It's easier to not use the `Optional` type here.  Just make this a `Meme` and then check whether or not it's `null` in line 36.  Also allows you to get rid of the extra `get()` in line 40.

Reply:
Should line 34 then be `Meme meme = memeMetadataService.findById(id).get();` then since the `findById` still returns an optional?